projects/shared/src/lib/form/protocol-field-set/protocol-field-set.component.html

```diff
@@ -0,0 +1,92 @@
+<div class="form-group" [formGroup]="form">
+  <label i18n="protocol label main@@protocolLabelMain">Harvest protocol *</label>
+  <div class="form-group-radios">
+    <label *ngIf="hasFileOption" i18n="protocol radio file@@protocolRadioFile">
+      <input type="radio" [formControlName]="protocolSwitchField" value="FILE" />
+      File upload
+    </label>
```

Comment:
We know we need it for the sandbox, and we know the sandbox is going to support more protocols eventually.  The sandbox is going to use this component.

Reply:
But that's the thing, we don't know this I think (it's merely possible). We don't know yet how the sandbox will support it. I'm not convinced even that we will reuse this component there. And I don't like adding dead functionality because we will forget about it if we decide to implement it some other way.

Reply:
Sandbox aside I think this refactor is positive - because it moves the validation logic out of a very crowded file.

We know the plan is to support the other protocols... as regards the "how" of showing them... well we know what the parameters are for HTTP/OAI-PMH imports are.  I don't think they'll change but if they do then the inputs that capture them can change.  Form field names can map to what's there.

Reply:
"I don't like adding dead functionality" either, but locally I have sandbox code that uses the file option - it'll be in place by the end of the week.